server/router/managers/cache-manager.js

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2021 Uber Technologies Inc.
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+const { ONE_HOUR_IN_MILLISECONDS } = require('../constants');
+
+class CacheManager {
+  constructor(cacheTimeLimit = ONE_HOUR_IN_MILLISECONDS) {
+    this.cache = null;
+    this.cacheExpiryDateTime = null;
+    this.cacheTimeLimit = cacheTimeLimit;
+  }
+
+  clearCache() {
+    this.cache = null;
+    this.cacheExpiryDateTime = null;
+  }
+
+  async get(fetchCallback) {
+    const { cache, cacheExpiryDateTime } = this;
+
+    if (cacheExpiryDateTime && Date.now() < cacheExpiryDateTime) {
+      return cache;
+    }
+
+    const data = await fetchCallback();
+
+    this.setCache(data);
+
+    return data;
+  }
```

Comment:
is it worth deduplicating concurrent `get`s?  I'm guessing there are multiple things that need the cluster cache, so when this expires it'll cause a sizable burst of concurrent updates and then multiple `set`s.
which is probably semantically fine - if the data rewinds a couple seconds due to a slow request doesn't really matter, and it very likely contains the same data anyway.  but it may not be desirable.

Reply:
Yea I was thinking something along these lines too. I think the reason I didn't handle this was that the request it makes is quite fast (relatively speaking) so there is not much lag anyway even if it fetches it a couple times. 

The only real reason why we are caching the request is to reduce hitting cadence server too much for data which doesn't change that often.